processors/src/utilities.cxx

```diff
@@ -93,87 +94,82 @@ Track* utils::buildTrack(EVENT::Track* lc_track,
     if (gbl_kink_data) {
         // Instantiate an LCRelation navigator which will allow faster access 
         // to GBLKinkData object
-        UTIL::LCRelationNavigator* gbl_kink_data_nav 
-            = new UTIL::LCRelationNavigator(gbl_kink_data);
-
+        std::shared_ptr<UTIL::LCRelationNavigator> gbl_kink_data_nav = std::make_shared<UTIL::LCRelationNavigator>(gbl_kink_data);
```

Comment:
This is fine. Being a really picky C++ nerd, this is a situation where std::unique_ptr would be better. (Replace shared with unique) The "std::unique_ptr" has slightly less overhead, and in this case, there is no other reference to the object.

Reply:
I take that back. We are still only at C++11 with Centos7, so std::unique_ptr<> doesn't exist (C++14 feature) and so we shouldn't use it.

Reply:
Hi Maurik,

As you correctly say, make_unique is only included in C++14. Let's keep it C++11 compatible. Additionally, we aren't in the need of code optimization at this stage.